conjure-go-client/httpclient/client.go

```diff
@@ -209,3 +213,12 @@ func (c *clientImpl) initializeRequestHeaders(ctx context.Context) http.Header {
 	}
 	return headers
 }
+
+func joinURIandPath(baseURI, reqPath string) (string, error) {
+	uri, err := url.Parse(baseURI)
+	if err != nil {
+		return "", werror.Wrap(err, "failed to parse request URL")
+	}
+	uri.Path = path.Join(uri.Path, reqPath)
```

Comment:
actually it looks like this is dropping trailing slashes, will have to make it slightly smarter